tests/integration_test/jwt_auth_integration_test.go

```diff
@@ -255,12 +255,12 @@ func TestAuthAllowMissing(t *testing.T) {
 			wantError:      `401 Unauthorized, {"code":401,"message":"Jwks remote fetch is failed"}`,
 		},
 		{
-			desc:           "Failed, token with wrong issuer",
+			desc:           "Successed, token with wrong issuer is considered as missing",
```

Comment:
Hi @qiwzhang, was there a change in upstream envoy that changes this test? Just curious why it changed, it's not obvious.

Reply:
Yes, it is due to my recently submitted pr https://github.com/envoyproxy/envoy/pull/13839
If a Jwt has "iss" and "iss" doesn't matched the one specified in the `provider`,  it will return JwtUnknownIssuer.   
In this PR, JwtUnkownIssuer is treated as JwtMissing. 

This is the part I am debating,   if the spec wants to verify JWT with "iss1"  and allow_missing.  But a JWT with "iss2" comes,   should we reject it?  Is this consider as "missing"?

For now,  it is considered as "missing" and it is accepted.

Reply:
I see, that is interesting. JWT requirement config allows both `allow_missing` and `allow_missing_or_failed`. Wouldn't the issuer mismatch be a verification failure? Then `allow_missing` should reject it, but `allow_missing_or_failed` should pass through.

https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/jwt_authn/v2alpha/config.proto#config-filter-http-jwt-authn-v2alpha-jwtrequirement

Reply:
Well, maybe you could clarify what the term "verification" means in this documentation. Verifying the entire token (including the issuer), or just verifying the signature matches?

Reply:
We verify signature, "aud" and "iss".